### PR TITLE
Add Lottery to Exiled

### PR DESCRIPTION
**Pull request by CharizardtheFireMage**
I can't confirm that this script is in working condition, so I highly suggest you test it before merging the PR.

### CODE REVIEW

**Comment by ghost** [comment]
whats the purpose of this if you haven't tested it yet?

File: chat-plugins/lottery.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		    allLotteryTickets.push(Db('tickets').object()[i]['Tickets'][v]);
+	    }
+	}
+};
+
+// Generates Tickets
+function generateTicket() {
+	let chars = 'abcdefghijklmnopqrstuvwxyz0123456789';
+	let lotteryTicket = '';
+	while (lotteryTicket.length < 8) {
```
Comment (ghost) [inline chat-plugins/lottery.js:28]: You could use the `UUID or shortid` modules for randomly generating ticket IDs.

File: chat-plugins/economy.js [hunk 2]
```diff
@@ -174,6 +175,9 @@ function handleBoughtItem(item, user, cost) {
 	} else if (item === 'profileteam') {
 		Db('hasteam').set(user);
 		this.sendReply('You can now set your team!');
+	} else if (item === 'ticket') {
+		let generatedTicket = giveTicket(user.userid);
```
Comment (ghost) [inline chat-plugins/economy.js:179]: Change this to `Exiled.giveTicket`

File: chat-plugins/lottery.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ * Lottery.js Made By Dragotic.
+ * A Lottery plugin that allows you to draw lottery and gain great rewards.
+ */
+
+'use strict';
+
+const color = require('../config/color');
+
+// Stores all the lottery tickets
+let allLotteryTickets = [];
```
Comment (ghost) [inline chat-plugins/lottery.js:12]: wouldn't having a single tickets object work the best? something like : 
`{
    'user1' : '[ticket 1, ticket 2]',
}`

File: chat-plugins/lottery.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+			target = toId(target);
+			if (!Db('tickets').has(target)) return this.errorReply(target + ' has no tickets.');
+			let userTickets = Db('tickets').object()[target]['Tickets'].join(', ');
+			this.sendReplyBox('<center><b><u>' + target + '\'s Tickets' + '</u></b></center><br />' + Db('tickets').object()[target]['Tickets'].join(', '));
+		},
+		give: function (target, room, user) {
+			if (!this.can('makechatroom')) return false;
+			if (!target) return this.parse('/help ticket');
+			let ticketId = giveTicket(toId(target));
+			this.sendReply(target + ' has been given a ticket with the id **' + ticketId + '**.');
```
Comment (ghost) [inline chat-plugins/lottery.js:90]: Add popups so users are informed they are getting tickets

File: chat-plugins/lottery.js (hunk 1 above)
Comment (Lord-Haji) [reply chat-plugins/lottery.js:28]: U can use let id = require('uuid/v4'); it's an offline lib so nothing to be added in package.json

File: chat-plugins/lottery.js (hunk 3 above)
Comment (Lord-Haji) [reply chat-plugins/lottery.js:12]: True

File: chat-plugins/lottery.js (hunk 4 above)
Comment (Lord-Haji) [reply chat-plugins/lottery.js:90]: Yes

File: chat-plugins/economy.js (hunk 2 above)
Comment (Lord-Haji) [reply chat-plugins/economy.js:179]: Yes else it's undefined here

**Comment by DeathlyPlays** [comment]
As requested

**Pull request closed without merging**